tests/vertexonly/test_vertex_only_fs.py

```diff
@@ -148,18 +148,22 @@ def functionspace_tests(vm):
     I2_io = Interpolator(2*TestFunction(V)*Constant(1, domain=vm), W)
     h2 = I2_io.interpolate(g)
     assert np.allclose(h2.dat.data_ro_with_halos[idxs_to_include], 2*np.prod(vm.input_ordering.coordinates.dat.data_ro_with_halos[idxs_to_include].reshape(-1, vm.input_ordering.geometric_dimension()), axis=1))
-    g = I_io.interpolate(h, transpose=True)
+
+    h_star = h.riesz_representation(riesz_map="l2")
+    g = I_io.interpolate(h_star, transpose=True)
     assert np.allclose(g.dat.data_ro_with_halos, np.prod(vm.coordinates.dat.data_ro_with_halos.reshape(-1, vm.geometric_dimension()), axis=1))
     with pytest.raises(NotImplementedError):
         # Can't use transpose on interpolators with expressions yet
-        g2 = I2_io.interpolate(h, transpose=True)
+        g2 = I2_io.interpolate(h_star, transpose=True)
```

Comment:
Here, and in loads of other places in the test suite, we are using `Interpolator.interpolate` in the way that we're about to warn everyone not to do - i.e. we're relying on getting a `Function` or `Cofunction` out rather than an expression. Surely we should switch to `assemble(Interpolate(...))` in these cases?

Reply:
In the specific case of testing out `Interpolator`s I guess you would need to do `assemble(Interpolator(...)._interpolate_future(...))`? Is that correct @nbouziani ?

Reply:
Wait that can't be right - in `__future__.py` you replace `Interpolator.interpolate` with `Interpolator._interpolate_future`. @nbouziani, either we haven't switched on `__future__.py` in all these tests or `Interpolate` somehow has a `.dat` associated with it. Which one is it?

Reply:
We import `interpolate` from `__future__.py`, so that tests align with the future interpolation behaviour. Then, we assemble any occurrence of `interpolate` or `Interpolator.interpolate`.

Reply:
But there's no assembly here... What am I missing?

Reply:
Oh this is outdated, nevermind